### PR TITLE
PR: Expand patterns to detect possible `site-packages` directories (Pythonpath manager)

### LINKED ISSUE #21228: Unable to import psycopg2 due to mixing packages from different Python installations
### Issue Report Checklist

* [x ] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [x] Tried basic troubleshooting (if a bug/error)
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
I have an M2 mac running Ventura. 
I am unable to import psycopg2 into Spyder. I installed spyder using the standalone MacOS installer. I manage the packages with pip and have added the path to PYTHONPATH manager in Spyder. I am using the python executable that came with the mac (3.9.6), though I tried spyder's included executable too. 
A number of packages were broken by the new mac chips. Usually you can get around it by installing with --no-binary

You can find lots of issues googling psycopg2 build errors with M1 macs. Generally the solution involves installing psycopg2-binary or installing some brew casks (libpq,openssl,postgresql) and then trying again to pip install psycopg2. This works, and if you successfully pip install either of these, you can open a python shell and import psycopg2.
However you can not import them into spyder. Both versions of psycopg2 give similar but distinct errors with the file `_psycopg.cpython-39-darwin.so`

### What steps reproduce the problem?

1. pip install psycopg2 or psycopg2-binary on M2 mac
2. import psycopg2 in spyder

### What is the expected output? What do you see instead?
Expected output would be a successful import
Current output is an ImportError with this file: `_psycopg.cpython-39-darwin.so`

### Paste Traceback/Error Below (if applicable)

```python-traceback

#if you have installed psycopg2-binary
import psycopg2

  File ~/Library/Python/3.9/lib/python/site-packages/psycopg2/__init__.py:51
    from psycopg2._psycopg import (                     # noqa

ImportError: dlopen(/Users/my_username/Library/Python/3.9/lib/python/site-packages/psycopg2/_psycopg.cpython-39-darwin.so, 0x0002): tried: '/Users/my_username/Library/Python/3.9/lib/python/site-packages/psycopg2/_psycopg.cpython-39-darwin.so' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/System/Volumes/Preboot/Cryptexes/OS/Users/my_username/Library/Python/3.9/lib/python/site-packages/psycopg2/_psycopg.cpython-39-darwin.so' (no such file), '/Users/my_username/Library/Python/3.9/lib/python/site-packages/psycopg2/_psycopg.cpython-39-darwin.so' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')))


#if you have installed psycopg2
import psycopg2

  File ~/Library/Python/3.9/lib/python/site-packages/psycopg2/__init__.py:51
    from psycopg2._psycopg import (                     # noqa

ImportError: dlopen(/Users/my_username/Library/Python/3.9/lib/python/site-packages/psycopg2/_psycopg.cpython-39-darwin.so, 0x0002): symbol not found in flat namespace '_PQbackendPID'

```

## Versions
5.4.4 and 6.0.0a1 (I tried both)

* Spyder version: 5.4.4 and 6.0.0a1 (I tried both)
* Python version: 3.9.6
* Operating System name/version: MacOS Ventura

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0                                                        :  1.4.1 (OK)
chardet >=2.0.0                                                             :  5.1.0 (OK)
cloudpickle >=0.5.0                                                         :  2.2.1 (OK)
cookiecutter >=1.6.0                                                        :  2.1.1 (OK)
diff_match_patch >=20181111                                                 :  20230430 (OK)
intervaltree >=3.0.2                                                        :  3.1.0 (OK)
IPython >=7.31.1,<9.0.0,!=8.8.0,!=8.9.0,!=8.10.0,!=8.11.0,!=8.12.0,!=8.12.1 :  8.14.0 (OK)
jedi >=0.17.2,<0.19.0                                                       :  0.18.2 (OK)
jellyfish >=0.7                                                             :  0.9.0 (OK)
jsonschema >=3.2.0                                                          :  4.17.3 (OK)
keyring >=17.0.0                                                            :  23.13.1 (OK)
nbconvert >=4.0                                                             :  7.6.0 (OK)
numpydoc >=0.6.0                                                            :  1.5.0 (OK)
parso >=0.7.0,<0.9.0                                                        :  0.8.3 (OK)
pexpect >=4.4.0                                                             :  4.8.0 (OK)
pickleshare >=0.4                                                           :  0.7.5 (OK)
psutil >=5.3                                                                :  5.9.5 (OK)
pygments >=2.0                                                              :  2.15.1 (OK)
pylint >=2.5.0,<3.0                                                         :  2.17.4 (OK)
pylint_venv >=3.0.2                                                         :  3.0.2 (OK)
pyls_spyder >=0.4.0                                                         :  0.4.0 (OK)
pylsp >=1.7.2,<1.8.0                                                        :  1.7.3 (OK)
pylsp_black >=1.2.0,<3.0.0                                                  :  1.3.0 (OK)
qdarkstyle >=3.0.2,<3.2.0                                                   :  3.1 (OK)
qstylizer >=0.2.2                                                           :  0.2.2 (OK)
qtawesome >=1.2.1                                                           :  1.2.3 (OK)
qtconsole >=5.4.2,<5.5.0                                                    :  5.4.3 (OK)
qtpy >=2.1.0                                                                :  2.3.1 (OK)
rtree >=0.9.7                                                               :  1.0.1 (OK)
setuptools >=49.6.0                                                         :  67.7.2 (OK)
sphinx >=0.6.6                                                              :  7.0.1 (OK)
spyder_kernels >=3.0.0b1,<3.0.0b2                                           :  3.0.0b1 (OK)
textdistance >=4.2.0                                                        :  4.5.0 (OK)
three_merge >=0.1.1                                                         :  0.1.1 (OK)
watchdog >=0.10.3                                                           :  3.0.0 (OK)
zmq >=22.1.0                                                                :  25.1.0 (OK)

# Optional:
cython >=0.21                                                               :  0.29.35 (OK)
matplotlib >=3.0.0                                                          :  3.7.1 (OK)
numpy >=1.7                                                                 :  1.25.0 (OK)
pandas >=1.1.1                                                              :  2.0.2 (OK)
scipy >=0.17.0                                                              :  1.10.1 (OK)
sympy >=0.7.3                                                               :  1.12 (OK)
```

### PR DESCRIPTION
**Pull request by ccordoba12**
## Description of Changes

* We were not detecting `site-packages` directories in Mac framework installations and two digit Python environments (e.g. Python 3.10).
* Add tests for the small utility function that does that job.

### Issue(s) Resolved

Fixes #21228.

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @ccordoba12

### CODE REVIEW

File: spyder/plugins/pythonpath/utils.py [hunk 1]
```diff
@@ -18,12 +18,16 @@
 def check_path(path):
     """Check that `path` is not a [site|dist]-packages folder."""
     if os.name == 'nt':
-        pat = re.compile(r'.*lib/(?:site|dist)-packages.*')
+        pattern = re.compile(r'.*(?:l|L)ib/(?:site|dist)-packages.*')
```
Comment (mrclary) [inline spyder/plugins/pythonpath/utils.py:21]: Is `?:` necessary? I don't think a non-capturing group is necessary in this context, but it doesn't hurt anything either...

File: spyder/plugins/pythonpath/utils.py [hunk 2]
```diff
@@ -18,12 +18,16 @@
...
     else:
-        pat = re.compile(r'.*lib/python.../(?:site|dist)-packages.*')
+        pattern = re.compile(
+            r'.*(:?lib|lib64)/'
+            '(?:python|python(\d+)|python(\d+)\.(\d+)|python(\d+)\.(\d+)(\d+))'
```
Comment (mrclary) [inline spyder/plugins/pythonpath/utils.py:25]:
```suggestion
            '(?:python|python\d+|python\d+\.\d+)'
```
I think the `()` are superfluous around `/d+`, and will just create unecessary groups. I think the pattern `python\d+\.\d+\d+` is redundant to `python\d+\.\d+`.

As with my previous comment, I don't think `?:` is necessary, but doesn't hurt either.

**Comment by ccordoba12** [comment]
Thanks @mrclary for your review. I'm not that good to work with regular expressions, so I really appreciate your help.

I applied your suggestions and checked that the new tests I added pass, so this should  be ready now.

**Review by mrclary — APPROVED** [review]
Looks good to me.

**Pull request merged**